### PR TITLE
Update RN version to 0.71

### PR DESCRIPTION
**Pull request by louiszawadzki**
### What does this PR do?

Update React Native version to 0.71

### Motivation

Be closer to the latest changes in API

### Review checklist (to be filled by reviewers)

- [x] Feature or bugfix MUST have appropriate tests
- [x] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [x] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)
- [x] If this PR is auto-generated, please make sure also to manually update the code related to the change

### CODE REVIEW

**Comment by louiszawadzki** [comment]
TODO: 
- Add test for React.createElement now that it's not the function used to create components
- merge mappers and solve conflicts

File: example/android/app/src/main/java/com/ddsdkreactnativeexample/MainActivity.java [hunk 1]
```diff
@@ -3,12 +3,11 @@
  * This product includes software developed at Datadog (https://www.datadoghq.com/).
  * Copyright 2016-Present Datadog, Inc.
  */
-
-package com.example.ddsdkreactnative;
+ 
+ package com.ddsdkreactnativeexample;
```
Comment (0xnm) [inline example/android/app/src/main/java/com/ddsdkreactnativeexample/MainActivity.java:7]: minor formatting suggestion

```suggestion

package com.ddsdkreactnativeexample;
```

File: example/android/app/build.gradle [hunk 2]
```diff
@@ -139,86 +94,13 @@ android {
...
 
-    compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
-    }
-
+    namespace "com.ddsdkreactnativeexample"
     defaultConfig {
-        applicationId "com.example.ddsdkreactnative"
+        applicationId "com.ddsdkreactnativeexample"
```
Comment (0xnm) [inline example/android/app/build.gradle:99]: maybe then `com.datadog.reactnativeexample`?

File: example/android/app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -1,3 +1,3 @@
 <resources>
-    <string name="app_name">DdSdkReactNative Example</string>
+    <string name="app_name">ddSdkReactnativeExample</string>
```
Comment (0xnm) [inline example/android/app/src/main/res/values/strings.xml:2]: I would propose changing to something more human-friendly:

```suggestion
    <string name="app_name">Datadog React Native SDK Example</string>
```

File: packages/codepush/src/__tests__/index.test.tsx [hunk 4]
```diff
@@ -101,7 +102,7 @@ describe('AppCenter Codepush integration', () => {
                 'env',
                 'appId',
                 true,
-                true,
+                false,
```
Comment (0xnm) [inline packages/codepush/src/__tests__/index.test.tsx:105]: is this change intentional?

File: packages/core/android/build.gradle [hunk 5]
```diff
@@ -78,77 +118,33 @@ repositories {
...
 def kotlin_version = getExtOrDefault('kotlinVersion')
 
 dependencies {
-  // noinspection GradleDynamicVersion
-  api 'com.facebook.react:react-native:+'
+  if (reactNativeMajorVersion == 0 && reactNativeMinorVersion < 71) {
+    // noinspection GradleDynamicVersion
+    api 'com.facebook.react:react-native:+'
+  } else {
+    if (project.getPluginManager().hasPlugin("com.facebook.react")) {
```
Comment (0xnm) [inline packages/core/android/build.gradle:141]: I will check this one, because I have a concern that `dependencies` closure here may be evaluated before plugins are applied, so that this condition _may always be_ evaluated to false

File: packages/codepush/src/__tests__/index.test.tsx (hunk 4 above)
Comment (louiszawadzki) [reply packages/codepush/src/__tests__/index.test.tsx:105]: If we enable resource tracking, we now have to provide a mock for XMLHttpRequest in the tests.
This is opens up for more reasons for these tests to break, and the codepush integration is not related at all to the resource tracking.
So I figured it was best to simply disable this option, as it does not impact the coverage of the tests and needs less code than maintaining an XMLHttpRequest mock

File: packages/core/android/build.gradle (hunk 5 above)
Comment (louiszawadzki) [reply packages/core/android/build.gradle:141]: It's true, we always go in the second branch

**Review by 0xnm — APPROVED** [review]
